### PR TITLE
Add Developer Guide Docs to MXNet Website

### PR DESCRIPTION
**Pull request by ys2843**
## Description ##
Add `Developer Guide` section on Docs & Tutorials" page above FAQ section. Please see preview.
@ehsanmok @TEChopra1000 @sandeep-krishnamurthy @aaronmarkham 

### Essentials ###
Please feel free to remove inapplicable items for your PR.
- [x] The PR title starts with [MXNET-$JIRA_ID], where $JIRA_ID refers to the relevant [JIRA issue](https://issues.apache.org/jira/projects/MXNET/issues) created (except PRs with tiny changes)
- [x] Changes are complete (i.e. I finished coding on this PR)
- [x] All changes have test coverage:

### Changes ###
- [x] Add developer guide

## Comments ##
- Preview: http://ec2-34-219-134-42.us-west-2.compute.amazonaws.com/

### CODE REVIEW

**Comment by ys2843** [comment]
@mxnet-label-bot add [website]

**Comment by ys2843** [comment]
@mxnet-bot run ci [unix-cpu]

File: docs/static_site/src/pages/api/developer_guide/1_github_contribution_and_PR_verification_tips.md [hunk 1]
```diff
@@ -0,0 +1,191 @@
...
+# GitHub contribution and PR verification tips 
+
+Here are some general git workflow tips
+
+## Setup and configure
+
+Fork [https://github.com/apache/incubator-mxnet](https://github.com/apache/incubator-mxnet) then
+
+```
+git clone https://github.com/your_username/incubator-mxnet
```
[the remark this answers is not in the document]
Comment (ys2843) [reply docs/static_site/src/pages/api/developer_guide/1_github_contribution_and_PR_verification_tips.md]: Done

**Comment by ehsanmok** [comment]
Thanks @ys2843 and @TEChopra1000 ! LGTM

**Pull request merged**